packages/pulse-core/tests/state.test.ts

```diff
@@ -1,8 +1,245 @@
 import Pulse from '../lib';
 
 const App = new Pulse();
-const MyState = App.State(true);
+const BooleanState = App.State<boolean>(true);
+const StringState = App.State<string>('Hello Pulse!');
+const ObjectState = App.State<{ days: Days }>({ days: { monday: true, wednesday: true } });
+const NullState = App.State(null);
```

Comment:
It it would be ideal if we used beforeAll / beforeEach here. I'll push an example to this file specifically, but if you could follow the same pattern with the others that would be fantastic!